File: 562backpackIV/backpackIV.java
public class Solution {
    /**
     * @param nums: an integer array and all positive numbers, no duplicates
     * @param target: An integer
     * @return: An integer
     */
    public int backPackIV(int[] nums, int target) {
        // write your code here
        final int dp[] = new int[target + 1];
        dp[0] = 1;
        Arrays.stream(nums).forEach(item -> java.util.stream.IntStream.range(item, target + 1).forEach(size -> dp[size] += dp[size - item]));
        return dp[dp.length - 1];
    }
}

File: 562backpackIV/backpackIV.js
/**
 * @param nums: an integer array and all positive numbers, no duplicates
 * @param target: An integer
 * @return: An integer
 */
const backPackIV = function (nums, target) {
    const dp = [1, ...Array(target).fill(0)]
    nums.forEach(item => {
        for (let size = 1; !Object.is(size, target + 1); ++size) dp[size] += size >= item ? dp[size - item] : 0})
    return dp[dp.length - 1]
}
